tests/test_linux/test_dhcp/test_parser.py

```diff
@@ -2,14 +2,20 @@
 
 from pyroute2.dhcp.dhcp4socket import AsyncDHCP4Socket
 from pyroute2.dhcp.enums import bootp, dhcp
+from pyroute2.dhcp.messages import ReceivedDHCPMessage
 
 
-def test_decode_simple_lease_process(pcap: PcapFile):
-    '''Decode a simple DHCP handshake using AsyncDHCP4Socket.'''
+def parse_pcap(
+    pcap: PcapFile, expected_packets: int
+) -> list[ReceivedDHCPMessage]:
     decoded_dhcp_messages = [AsyncDHCP4Socket._decode_msg(i) for i in pcap]
```

Comment:
Just wondering, how does it work with pcap header.

See also: https://github.com/svinota/pyroute2/blob/3d0d36f9c42c80dca8458e8ab072c46d6adc33fd/pyroute2/decoder/loader.py#L258-L266

Reply:
Oh, i had not noticed it ! I put together a quick test fixture without seeing it, i'll see how i can use this code instead

Reply:
If you need any changes in the decoder, pls let me know or feel free to fix

(and I have to document this part at least)